pkg/ruler/api.go

```diff
@@ -168,20 +177,20 @@ func (a *API) PrometheusRules(w http.ResponseWriter, req *http.Request) {
 		grp := RuleGroup{
 			Name:           g.Group.Name,
 			File:           g.Group.Namespace,
-			Rules:          make([]rule, len(g.ActiveRules)),
+			Rules:          []rule{},
```

Comment:
It doesn't feel like a good change - not filtering will happen less often IMO (as again, I don't think clients use this feature on the general case) so I'd keep the hint to at most active rules.

Reply:
Those changes are now reverted since the filtering is done on each ruler